docs/docs/dev-setup.md

```diff
@@ -4,9 +4,189 @@ title: Basic Setup
 sidebar_label: Basic Setup
 ---
 
-1. Install `west` from pip.
-1. Install the ARM SDK from Zephyr project.
-1. (Optional) Install 32-bit libs for building and testing locally.
-1. `west init -l zmk`
-1. Build it.
-1. Flash it.
+## Prerequisites
+
+A unix-like environment with the following base packages installed:
+
+- Git
+- Python 3
+- `pip`
+- `wget`
+- devicetree compiler
+- CMake
+- `dfu-util`
+- Various build essentials, e.g. gcc, automake, autoconf
+
+### Debian/Ubuntu
+
+On Debian and Ubuntu, this can be accomplished with:
+
+```bash
+apt-get install -y \
+    git \
+    wget \
+    autoconf \
+	automake \
+	build-essential \
+	ccache \
+	device-tree-compiler \
+	dfu-util \
+	g++ \
+	gcc \
+	gcc-multilib \
+	libtool \
+	make \
+    cmake \
+	python3-dev \
+	python3-pip \
+	python3-setuptools \
+	xz-utils
+```
+
+### Fedora
+
+TODO
+
+### macOS
+
+TODO
+
+### WSL
+
+TODO
+
+## Setup
+
+### West Build Command
+
+`west` is the [Zephyr™ meta-tool](https://docs.zephyrproject.org/latest/guides/west/index.html) used to configure and build Zephyr™ applications. It can be installed by using the `pip` python package manager:
+
+```bash
+pip3 install --user west
+```
+
+:::note
+If you don't already have it configured, you may need to update your
+`PATH` to include the pip install path. See [User Installs](https://pip.pypa.io/en/stable/user_guide/#user-installs) and [Stack Overflow](https://stackoverflow.com/questions/38112756/how-do-i-access-packages-installed-by-pip-user) for more details.
+:::
+
+### Zephyr™ ARM SDK
+
+To build firmwares for the ARM architecture (all supported MCUs/keyboards at this point), you'll need to install the Zephyr™ ARM SDK to your system:
+
+```
+export ZSDK_VERSION=0.11.2
+wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${ZSDK_VERSION}/zephyr-toolchain-arm-${ZSDK_VERSION}-setup.run" && \
```

Comment:
I had to do sudo on all of these which was kinda awkward with &&s as well. It's fine, but even tripped me up for a second.